src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WidgetPageObject.java

```diff
@@ -88,43 +106,56 @@ public boolean areAllValidSwappedForIFrames() {
     return result;
   }
 
+  /**
+   * Make sure that all tags defined in the widget tags field are loaded.
+   */
   public boolean areLoaded() {
     boolean result = true;
-    int i = 1;
-    List<WebElement> widgetIFrameList = getWidgetIFrameList();
-
-    if (widgetIFrameList.isEmpty()) {
-      result = false;
-    } else {
-      for (WebElement widgetIFrame : widgetIFrameList) {
-        if (!isWidgetVisible(widgetIFrame, getWidgetBody())) {
-          result = false;
-          PageObjectLogging.log(getTagName() + " #" + i, MercuryMessages.VISIBLE_MSG, result);
-          break;
-        }
-        PageObjectLogging.log(getTagName() + " #" + i++, MercuryMessages.VISIBLE_MSG, result);
+    String[] tags = getTags();
+    for (int i = 0; i < tags.length; i++) {
+      if (!isWidgetVisible(i)) {
+        result = false;
+        PageObjectLogging.log(getTagName() + " #" + i, MercuryMessages.VISIBLE_MSG, result);
+        break;
       }
+      PageObjectLogging.log(getTagName() + " #" + i++, MercuryMessages.VISIBLE_MSG, result);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)